src/main/java/org/opentripplanner/routing/impl/OrcaFareServiceImpl.java

```diff
@@ -0,0 +1,306 @@
+        SOUND_TRANSIT,
+        WASHINGTON_STATE_FERRIES
+    }
+
+    private static final Map<String, Function<Route, RideType>> classificationStrategy = new HashMap<>();
+
+    // If set to true, the test ride price is used instead of the actual agency cash fare.
+    public boolean IS_TEST;
+    public static final float DEFAULT_TEST_RIDE_PRICE = 3.49f;
+
+    public OrcaFareServiceImpl(Collection<FareRuleSet> regularFareRules) {
+        addFareRules(Fare.FareType.regular, regularFareRules);
+        addFareRules(Fare.FareType.senior, regularFareRules);
+        addFareRules(Fare.FareType.youth, regularFareRules);
+        addFareRules(Fare.FareType.electronicRegular, regularFareRules);
+        addFareRules(Fare.FareType.electronicYouth, regularFareRules);
+        addFareRules(Fare.FareType.electronicSpecial, regularFareRules);
+        addFareRules(Fare.FareType.electronicSenior, regularFareRules);
+
+        classificationStrategy.put(
+            COMM_TRANS_AGENCY_ID,
+            routeData -> {
+                try {
+                    int routeId = Integer.parseInt(routeData.getShortName());
+                    if (routeId >= 400 && routeId <= 899) {
+                        return RideType.COMM_TRANS_COMMUTER_EXPRESS;
+                    }
+                    return RideType.COMM_TRANS_LOCAL_SWIFT;
+                } catch (NumberFormatException e) {
+                    LOG.warn("Unable to determine comm trans route id from {}.", routeData.getShortName(), e);
+                    return RideType.COMM_TRANS_LOCAL_SWIFT;
+                }
+            }
+        );
+        classificationStrategy.put(
+            KC_METRO_AGENCY_ID,
+            routeData -> {
+                if (routeData.getType() == ROUTE_TYPE_FERRY &&
+                    routeData.getDesc().contains("Water Taxi: West Seattle")) {
+                    return RideType.KC_WATER_TAXI_WEST_SEATTLE;
+                } else if (routeData.getType() == ROUTE_TYPE_FERRY &&
+                    routeData.getDesc().contains("Water Taxi: Vashon Island")) {
+                    return RideType.KC_WATER_TAXI_VASHON_ISLAND;
+                }
+                return RideType.KC_METRO;
+            }
+        );
+        classificationStrategy.put(SOUND_TRANSIT_AGENCY_ID, routeData -> RideType.SOUND_TRANSIT);
+        classificationStrategy.put(EVERETT_TRANSIT_AGENCY_ID, routeData -> RideType.EVERETT_TRANSIT);
+        classificationStrategy.put(PIERCE_COUNTY_TRANSIT_AGENCY_ID, routeData -> RideType.PIERCE_COUNTY_TRANSIT);
+        classificationStrategy.put(SKAGIT_TRANSIT_AGENCY_ID, routeData -> RideType.SKAGIT_TRANSIT);
+        classificationStrategy.put(SEATTLE_STREET_CAR_AGENCY_ID, routeData -> RideType.SEATTLE_STREET_CAR);
+        classificationStrategy.put(WASHINGTON_STATE_FERRIES_AGENCY_ID, routeData -> RideType.WASHINGTON_STATE_FERRIES);
+        classificationStrategy.put(KITSAP_TRANSIT_AGENCY_ID, routeData -> RideType.KITSAP_TRANSIT);
```

Comment:
Originally, I saw thought we might be able to get away with not expanding Kitsap's ride types. But in order to properly capture the reduced fares on their fast ferry service, I think there need to be 3 Kitsap Ride types. One of them for regular service and another for fast ferry in each direction. See [Kitsap fares](https://www.kitsaptransit.com/fares/fares).